iothub/service/src/Registry2/ConfigurationsClient.cs

```diff
@@ -0,0 +1,460 @@
+    /// </summary>
+    public class ConfigurationsClient
+    {
+        private const string ConfigurationRequestUriFormat = "/configurations/{0}?{1}";
+        private const string ConfigurationsRequestUriFormat = "/configurations/?top={0}&{1}";
+        private const string ApplyConfigurationOnDeviceUriFormat = "/devices/{0}/applyConfigurationContent?" + ClientApiVersionHelper.ApiVersionQueryString;
+        private static readonly TimeSpan s_defaultOperationTimeout = TimeSpan.FromSeconds(100);
+
+        private IHttpClientHelper _httpClientHelper;
+
+        /// <summary>
+        /// Creates an instance of this class. Provided for unit testing purposes only.
+        /// </summary>
+        protected ConfigurationsClient()
+        {
+        }
+
+        internal ConfigurationsClient(IotHubConnectionProperties connectionProperties, HttpTransportSettings transportSettings)
+        {
+            _httpClientHelper = new HttpClientHelper(
+                connectionProperties.HttpsEndpoint,
+                connectionProperties,
+                ExceptionHandlingHelper.GetDefaultErrorMapping(),
+                s_defaultOperationTimeout,
+                transportSettings.Proxy,
+                transportSettings.ConnectionLeaseTimeoutMilliseconds);
+        }
+
+        /// <summary>
+        /// Creates a new Configuration for Azure IoT Edge in IoT hub
+        /// </summary>
+        /// <param name="configuration">The Configuration object being created.</param>
+        /// <returns>The Configuration object.</returns>
+        /// <seealso href="https://docs.microsoft.com/azure/iot-hub/iot-hub-automatic-device-management"/>
+        public virtual Task<Configuration> CreateAsync(Configuration configuration)
+        {
+            return CreateAsync(configuration, CancellationToken.None);
+        }
+
+        /// <summary>
+        /// Creates a new Configuration for Azure IoT Edge in IoT hub
+        /// </summary>
+        /// <param name="configuration">The Configuration object being created.</param>
+        /// <param name="cancellationToken">The token which allows the operation to be canceled.</param>
+        /// <returns>The Configuration object.</returns>
+        /// <seealso href="https://docs.microsoft.com/azure/iot-hub/iot-hub-automatic-device-management"/>
+        public virtual Task<Configuration> CreateAsync(Configuration configuration, CancellationToken cancellationToken)
+        {
+            if (Logging.IsEnabled)
+                Logging.Enter(this, $"Adding configuration: {configuration?.Id}", nameof(CreateAsync));
+
+            try
+            {
+                EnsureInstanceNotClosed();
+
+                if (!string.IsNullOrEmpty(configuration.ETag))
+                {
+                    throw new ArgumentException(ApiResources.ETagSetWhileCreatingConfiguration);
+                }
+
+                var errorMappingOverrides = new Dictionary<HttpStatusCode, Func<HttpResponseMessage, Task<Exception>>>
```

Comment:
Go ahead and delete all these errorMappingOverrides for now. Later, we'll address the thrown exception structure for the service client as a whole so that we don't need to have overrides like this.

Reply:
added the changes